src/cloud-api-adaptor/docker/README.md

```diff
@@ -0,0 +1,132 @@
+# Introduction
+
+The `docker` provider simulates a pod VM inside a docker container.
+
+> **Note**: Run the following commands from the root of this repository.
+
+## Prerequisites
+
+- Install `docker` [engine](https://docs.docker.com/engine/install/)
```

Comment:
Do we need to have set up the docker user group to run it as a non-root user (https://docs.docker.com/engine/install/linux-postinstall/). I had a whole bunch of permissions issues (detailed below) and that might have been part of the fix.

I also had to install a version newer that the default ubuntu 22.04 apt package to get docker 1.44+ api, so that might need to be stated?

Reply:
I'll add these in the README